vite-pwa.config.ts

```diff
@@ -1,12 +1,16 @@
-import { mergeConfig, defineConfig } from 'vite'
+import { defineConfig, mergeConfig } from 'vite'
 import { VitePWA } from 'vite-plugin-pwa'
 
 import viteBaseConfig from './vite-base.config'
 import { manifest } from './vite-config/manifest'
 
-export default mergeConfig(
-  viteBaseConfig,
-  defineConfig({
+export default defineConfig(() => {
+  /** @see https://vite.dev/guide/api-javascript.html#mergeconfig */
+  return mergeConfig(viteBaseConfig, {
+    /** @see https://vite.dev/config/shared-options.html#define */
+    define: {
+      __VITE_ROUTER_HISTORY_MODE__: JSON.stringify(process.env.VITE_ROUTER_HISTORY_MODE)
+    },
```

Comment:
All the changes in that file will affect only the PWA. We have also Android and Electron configurations that extends the base config https://github.com/Adamant-im/adamant-im/blob/dev/vite-base.config.ts

1. Move the changes to `vite-base.config.ts`
2. Use same approach as for `VITE_PUBLIC_PATH`. I don't think we should to use `define` here.

Reply:
Do we really need hash routing for Android and Electron?

Reply:
For Electron yes. For other builds, excluding this one, no. We can control the behaviour with `VITE_ROUTER_HISTORY_MODE` anyway.

Reply:
As we resolved [issue](https://github.com/Adamant-im/adamant-im/pull/778#discussion_r2061472958), perhaps this workaround only needed for Electron builds